docs/draft/rest_api.md

```diff
@@ -1,201 +1,324 @@
+    },
+    "cloudformation": {
+        "href":"/platforms/cloudformation"
+    }
+]
 ```
-- *id*: The id of the transformation (unsigned Integer)
-- *result*: the URL path (on the HTTP server) to download the result archive. If the transformation is still in progress or has errored this value will be null
+NOTE: Later it is possible to add platform specific preferences here
 
-### Getting the status of a transformation
-```
-GET transformation/{id}/status
-```
+### Manipulating TOSCA Models
+
+##### GET /toscamodels
+Get a list of all toscamodels.  
 
-*Parameters*:
-- **id**: The id of the transformation for which the result should be retrieved.
+*returns*:
+```json
+[
+    {
+        "href":"/toscamodels/hello-world",
+        "name":"hello-world",
+        "csar": {
+            "href":"/toscamodels/hello-world/csar"
+        },
+        "transformations": {
+            "href":"/toscamodels/hello-world/transformations"
+        }
+    },
+    {
+        "href":"/toscamodels/billing-app",
+        "name":"billing-app",
+        "csar": {
+            "href":"/toscamodels/billing-app/csar"
+        },
+        "transformations": {
+            "href":"/toscamodels/billing-app/transformations"
+        }
+    }
+]
+```
+##### POST /toscamodels
+Create a new TOSCA model. Returns a link to the new resource.
 
-**on success returns:**
-  
-HTTP Response Code: 200
+*Request body:*  
 ```json
 {
-  "id": 1,
-  "status": "done",
-  "progress":  "100",
-  "message": "Tranformation was sucessful."
+    "name":"{appName}"
 }
 ```
-- *status*: returns the current status. Possible values are `user-input`, `queued`, `started`, `stopped`, `failed` or `done`
-- *progress*: returns the progress as percentage from 0-100% (Unsigned Integer, range 0  to 100)
-- *message*: String message to quickly inform the user about whats happening.
+- name: name (String) of the application used for representation. Must be unique. Allowed characters: [a-z0-9_-]. Upper case letters are automatically converted to lower case.
 
-### Retrieve Logs for a transformation
+*returns*: `201 - created`
 ```
-GET transformation/{id}/log/{linecount}
+{
+    "href":"/toscamodels/billing-app",
+    "name":"billing-app",
+    "csar": {
+        "href":"/toscamodels/billing-app/csar"
+    },
+    "transformations": {
+        "href":"/toscamodels/billing-app/transformations"
+    }
+}
 ```
- 
-*Parameters*:
-- *id*: The id of the transformation for which the logs should be retrieved
-- *linecount*: (Optional, 50 by default) The Maximum linecount to return. if this parameter is ``all`` all lines of the log will be sent
 
-**on success returns:**
+*ERRORS*:  
+`422` - `name` value already in use by other toscamodel
+
+##### DELETE /toscamodels
+Delete all TOSCA models
 
-This call returns the `'linecount'` last lines of the log. 
+##### GET /toscamodels/{appName}
+Get the TOSCA model which name matches {appName}.
 
-HTTP Response Code: 200
-  
+*returns*:
 ```json
 {
```

Comment:
I would at a id-field our web app or cli might need ids to handle the transformations. Angular for example could take ids to optimize viewing lists etc.

Reply:
appName would serve as id field since its an unique field. Having an id field would cause unnecessary bloat of data, IMO

Reply:
Regarding filtering and limiting data: We might need this later, but I would only add it when we see the need for it --  see the danger of bloating the api with overengineered functions

Reply:
👍